smartsim/control/controller.py

```diff
@@ -352,6 +360,85 @@ def _launch_orchestrator(self, orchestrator):
         self._save_orchestrator(orchestrator)
         logger.debug(f"Orchestrator launched on nodes: {orchestrator.hosts}")
 
+    def _launch_ray_cluster(self, ray_cluster):
+        """Launch a Ray Cluster instance
+
+        This function will launch the Ray Cluster instance and
+        if on WLM, find the nodes where it was launched and
+        set them in the JobManager
+
+        :param orchestrator: ray cluster to launch
+        :type orchestrator: RayCluster
+        """
+        # if the Ray cluster was launched as a batch workload
+        if ray_cluster.batch:   
+            head_batch_step = self._create_batch_job_step(ray_cluster.head_model)
+            self._launch_step(head_batch_step, ray_cluster.head_model)
+            ray_cluster._get_ray_head_node_address()
+            try:
+                nodelist = self._launcher.get_step_nodes([head_batch_step.name])
+                ray_cluster._hosts = nodelist[0]
+                
+            # catch if it fails or launcher doesn't support it
+            except LauncherError:
+                logger.debug("WLM Ray head node aquisition failed")
+            except SSUnsupportedError:
+                logger.debug("WLM Ray head node acquisition unsupported")
+                
+            if ray_cluster.worker_model:
+                ray_cluster._update_worker_model()
+                worker_batch_step = self._create_batch_job_step(ray_cluster.worker_model)
+                self._launch_step(worker_batch_step, ray_cluster.worker_model)
+                try:
+                    nodelist = self._launcher.get_step_nodes([worker_batch_step.name])
+                    ray_cluster._hosts.extend(nodelist[0])
+                    # catch if it fails or launcher doesn't support it
+                except LauncherError:
+                    logger.debug("WLM Ray worker node aquisition failed")
+                except SSUnsupportedError:
+                    logger.debug("WLM Ray worker node acquisition unsupported")
+        else:
+            head_step = self._create_job_step(ray_cluster.head_model)
+            self._launch_step(head_step, ray_cluster.head_model)
+            ray_cluster._get_ray_head_node_address()
+            if isinstance(self._launcher, SlurmLauncher):
```

Comment:
Why do we specify slurm here? because slurm is the only one with node acquisition implemented? I would think we should leave this conditional out and let the ``WLMLauncher`` class throw a ``SSUnsupportedError``. 

That way when we do find mechanisms to parse node information from PBS and/or cobalt, adding them requires less refactoring.

Reply:
OK, I like this.